### DISCUSSION TITLE
'Itemfilter' for CMI

### DISCUSSION THREAD

**Discussion by VimaMT** [Zrips/CMI#3493] [Feature Requests]
**Description of issue or feature request: Add /itemfilter GUI to CMI**

Several months ago one of the servers I played on had an /itemfilter command which brought up a GUI in which players could drop items that they did not want to pick up while they dropped them from mining, digging, or cutting. They would just leave those items on the ground behind them.  Please consider such a feature for CMI.

**Comment by CloudeLecaw** [comment]
you mean a itempickup filter. So when they put in Cobble they dont picup cobble right?

Nice idea, it would be nice if you could make some idea how todesign this:)
If u want i can make a suggestion :D

**_'Itemfilter'Pickupflter  for CMI for players_ by world**
Players using command:  the command could be: **_/cmi personalfilter (playerName) (world)_**
the argument "playerName" is optinal for admins to see/edit others filter.
the argument "world" is optinal for admins to see/edit their/others filter they defined in other worlds
permissions:  **_cmi.command.persinalfilter_**   and **_cmi.command.personalfilter.others_**

How to use:
1. use command:  **/cmi peronsalfilter**
![Snag_cbdf19](https://user-images.githubusercontent.com/12080017/84567814-77c1e580-ad7b-11ea-9d1c-2c636abdfea9.png)

2. a ChestGui opens where player can putit items
![image](https://user-images.githubusercontent.com/12080017/84567903-33831500-ad7c-11ea-86f7-9d4da70ac182.png)

3. Now player can start mining or what ever

Result: item from the filter where not picked up for the player
![Snag_d776ef](https://user-images.githubusercontent.com/12080017/84568067-3c281b00-ad7d-11ea-998c-ceb83f221a14.png)

![Snag_d9d5fd](https://user-images.githubusercontent.com/12080017/84568118-9923d100-ad7d-11ea-9405-f8187f11a29e.png)

The Filter allways only is setup for the world their are currently inside! So each world where they want a filter they need to setup one while be in this world.

Note:
The Filter should be able to be saved in mysql database too (by player and by world) + a list in config where the owner can define worlds where the filter is enabled)
This also needs a deny meaaasge like "Filtering Items is not allowed for this world" when a world is not enabled by the config. In all worlds not defined in config there cant be filters defined + admins cant the worldname as worldagrgument. 

All should be have Tabcompletation: this means - the playernames (argument1) and also the enabled worldnames (argrument2)

**Comment by VimaMT** [comment]
I think you have a good grasp of what I am looking for. I was not thinking of anything world-specific, but I would not object.  😀
